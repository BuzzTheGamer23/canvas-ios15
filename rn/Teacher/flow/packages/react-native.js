// @flow
export type PanResponderGestureState = {
  stateID: number,
  moveX: number,
  moveY: number,
  x0: number,
  y0: number,
  dx: number,
  dy: number,
  vx: number,
  vy: number,
}
