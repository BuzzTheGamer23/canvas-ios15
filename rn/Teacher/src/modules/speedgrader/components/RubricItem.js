// @flow

import React, { Component } from 'react'
import {
  View,
  Image,
  StyleSheet,
  AlertIOS,
} from 'react-native'
import i18n from 'format-message'
import { Text } from '../../../common/text'
import { CircleToggle, LinkButton } from '../../../common/buttons'
import Images from '../../../images'

export default class RubricItem extends Component {
  props: RubricItemProps
  state: RubricItemState

  constructor (props: RubricItemProps) {
    super(props)

    let grade = this.props.grade && this.props.grade.points

    this.state = {
      selectedOption: grade,
    }
  }

  showDescription = () => {
    this.props.showDescription(this.props.rubricItem.id)
  }

  changeSelected = (value: number) => {
    this.setState({ selectedOption: value })
    this.props.changeRating(this.props.rubricItem.id, value)
  }

  isCustomGrade = () => this.props.rubricItem.ratings.every(({ points }) => points !== this.state.selectedOption) && this.state.selectedOption != null

  promptCustom = (customMessage: ?string) => {
    AlertIOS.prompt(
      i18n('Customize Grade'),
      customMessage,
      (value) => {
        value = +value
        if (isNaN(value)) {
          this.promptCustom(i18n('Please enter a number'))
        }

        this.changeSelected(+value)
      },
      'plain-text',
      this.isCustomGrade() ? String(this.state.selectedOption) : '',
      'number-pad'
    )
  }

  render () {
    let { rubricItem } = this.props
    let isCustomGrade = this.isCustomGrade()
    return (
      <View style={styles.container}>
        <Text style={styles.description}>{rubricItem.description}</Text>
        <View style={styles.ratings}>
          {rubricItem.ratings.slice().reverse().map(rating => (
            <CircleToggle
              key={rating.id}
              style={styles.circle}
              on={this.state.selectedOption === rating.points}
              value={rating.points}
              onPress={this.changeSelected}
            >
              {rating.points}
            </CircleToggle>
          ))}
          <CircleToggle
            key='add'
            style={styles.circle}
            on={isCustomGrade}
            value={isCustomGrade ? this.state.selectedOption : ''}
            onPress={this.promptCustom}
            accessibilityLabel={i18n('Customize Grade')}
          >
            { isCustomGrade
              ? this.state.selectedOption
              : <Image source={Images.add} />
            }
          </CircleToggle>
        </View>
        <View style={styles.buttons}>
          <LinkButton style={styles.button}>{i18n('Add Comment')}</LinkButton>
          <LinkButton
            style={styles.button}
            onPress={this.showDescription}
            testID='rubric-item.description'
          >
            {i18n('View long description')}
          </LinkButton>
        </View>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    paddingTop: 16,
    paddingBottom: 8,
    paddingHorizontal: 16,
    overflow: 'hidden',
  },
  description: {
    fontWeight: '600',
  },
  ratings: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    marginTop: 12,
    marginBottom: 8,
  },
  circle: {
    marginRight: 8,
    marginBottom: 8,
  },
  buttons: {
    flexDirection: 'row',
  },
  button: {
    fontSize: 14,
    fontWeight: '500',
    paddingRight: 16,
  },
})

type RubricItemProps = {
  rubricItem: Rubric,
  grade: RubricAssessment,
  showDescription: (string) => void,
  changeRating: (string, number) => void,
}

type RubricItemState = {
  selectedOption: ?number,
}
