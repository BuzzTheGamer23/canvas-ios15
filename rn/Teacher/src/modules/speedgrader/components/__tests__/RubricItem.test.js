// @flow

import React from 'react'
import { AlertIOS, NativeModules, ActionSheetIOS } from 'react-native'
import RubricItem from '../RubricItem'
import renderer from 'react-test-renderer'
import explore from '../../../../../test/helpers/explore'

jest.mock('../../../../common/components/CircleToggle', () => 'CircleToggle')
jest.mock('react-native-button', () => 'Button')
jest.mock('AlertIOS', () => ({
  prompt: jest.fn(),
}))
jest.mock('ActionSheetIOS', () => ({
  showActionSheetWithOptions: jest.fn(),
}))

const templates = {
  ...require('../../../../api/canvas-api/__templates__/rubric'),
}

let defaultProps = {
  rubricItem: templates.rubric(),
  grade: { comments: '' },
  showDescription: jest.fn(),
  changeRating: jest.fn(),
  openCommentKeyboard: jest.fn(),
  deleteComment: jest.fn(),
}

describe('RubricItem', () => {
  beforeEach(() => jest.resetAllMocks())

  it('renders', () => {
    let tree = renderer.create(
      <RubricItem {...defaultProps} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders an already selected grade', () => {
    let props = {
      ...defaultProps,
      grade: {
        points: 10,
        comments: '',
      },
    }
    let tree = renderer.create(
      <RubricItem {...props} />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders the selected custom grade', () => {
    let props = {
      ...defaultProps,
      grade: {
        points: 12,
        comments: '',
      },
    }
    let tree = renderer.create(
      <RubricItem {...props} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('calls showDescription when the button is pressed', () => {
    let tree = renderer.create(
      <RubricItem {...defaultProps} />
    ).toJSON()

    let button = explore(tree).selectByID('rubric-item.description') || {}
    button.props.onPress()

    expect(defaultProps.showDescription).toHaveBeenCalledWith(defaultProps.rubricItem.id)
  })

  it('changes the currently selected value when a circle is pressed', () => {
    let tree = renderer.create(
      <RubricItem {...defaultProps} />
    )

    let button = explore(tree.toJSON()).selectByID(`rubric-item.points-${defaultProps.rubricItem.id}`) || {}
    button.props.onPress()

    expect(tree.toJSON()).toMatchSnapshot()
  })

  it('calls changeRating with the id and value', () => {
    let tree = renderer.create(
      <RubricItem {...defaultProps} />
    ).toJSON()

    let button = explore(tree).selectByID(`rubric-item.points-${defaultProps.rubricItem.ratings[0].id}`) || {}
    button.props.onPress(0)

    expect(defaultProps.changeRating).toHaveBeenCalledWith('2', 0)
  })

  it('gets the value from prompting for a custom value', () => {
    let tree = renderer.create(
      <RubricItem {...defaultProps} />
    ).toJSON()

    let button = explore(tree).selectByProp('testID', `rubric-item.customize-grade-${defaultProps.rubricItem.id}`).pop()
    button.props.onPress()

    expect(AlertIOS.prompt).toHaveBeenCalled()
    AlertIOS.prompt.mock.calls[0][2][1].onPress('12')
    expect(defaultProps.changeRating).toHaveBeenCalledWith(defaultProps.rubricItem.id, 12)
    expect(NativeModules.NativeAccessibility.focusElement).toHaveBeenCalledWith(`rubric-item.customize-grade-${defaultProps.rubricItem.id}`)
  })

  it('refocuses the customize button on cancel of the prompt', () => {
    let tree = renderer.create(
      <RubricItem {...defaultProps} />
    ).toJSON()

    let button = explore(tree).selectByProp('testID', `rubric-item.customize-grade-${defaultProps.rubricItem.id}`).pop()
    button.props.onPress()

    expect(AlertIOS.prompt).toHaveBeenCalled()
    AlertIOS.prompt.mock.calls[0][2][0].onPress()
    expect(NativeModules.NativeAccessibility.focusElement).toHaveBeenCalledWith(`rubric-item.customize-grade-${defaultProps.rubricItem.id}`)
  })

  it('will call prompt with a default value if there is an existing custom grade', () => {
    let props = {
      ...defaultProps,
      grade: {
        points: 1234,
        comments: '',
      },
    }
    let tree = renderer.create(
      <RubricItem {...props} />
    ).toJSON()

    let button = explore(tree).selectByProp('testID', `rubric-item.customize-grade-${defaultProps.rubricItem.id}`).pop()
    button.props.onPress()

    expect(AlertIOS.prompt.mock.calls[0][4]).toEqual('1234')
  })

  it('will call openCommentKeyboard when the add comment button is pressed', () => {
    let tree = renderer.create(
      <RubricItem {...defaultProps} />
    ).toJSON()

    let button = explore(tree).selectByID(`rubric-item.add-comment-${defaultProps.rubricItem.id}`) || {}
    button.props.onPress()

    expect(defaultProps.openCommentKeyboard).toHaveBeenCalledWith(defaultProps.rubricItem.id)
  })

  it('will not show the add comment button and will show the comment if there is one', () => {
    let props = {
      ...defaultProps,
      grade: { comments: 'A comment' },
    }

    let tree = renderer.create(
      <RubricItem {...props} />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('Will open an action sheet when a rubric comment is pressed', () => {
    let props = {
      ...defaultProps,
      grade: { comments: 'A comment' },
    }

    let tree = renderer.create(
      <RubricItem {...props} />
    ).toJSON()

    let button = explore(tree).selectByID(`rubric-item.edit-comment-${defaultProps.rubricItem.id}`) || {}
    button.props.onPress()

    expect(ActionSheetIOS.showActionSheetWithOptions).toHaveBeenCalled()
  })

  it('will do nothing when cancel is pressed in the edit action sheet', () => {
    let props = {
      ...defaultProps,
      grade: { comments: 'A comment' },
    }

    let tree = renderer.create(
      <RubricItem {...props} />
    ).toJSON()

    let button = explore(tree).selectByID(`rubric-item.edit-comment-${defaultProps.rubricItem.id}`) || {}
    button.props.onPress()

    ActionSheetIOS.showActionSheetWithOptions.mock.calls[0][1](2)
    expect(defaultProps.deleteComment).not.toHaveBeenCalled()
    expect(defaultProps.openCommentKeyboard).not.toHaveBeenCalled()
  })

  it('will call deleteComment when the delete option is pressed in the edit action sheet', () => {
    let props = {
      ...defaultProps,
      grade: { comments: 'A comment' },
    }

    let tree = renderer.create(
      <RubricItem {...props} />
    ).toJSON()

    let button = explore(tree).selectByID(`rubric-item.edit-comment-${defaultProps.rubricItem.id}`) || {}
    button.props.onPress()

    ActionSheetIOS.showActionSheetWithOptions.mock.calls[0][1](1)
    expect(defaultProps.deleteComment).toHaveBeenCalledWith(defaultProps.rubricItem.id)
  })

  it('will call openCommentKeyboard when the edit option is pressed in the edit action sheet', () => {
    let props = {
      ...defaultProps,
      grade: { comments: 'A comment' },
    }

    let tree = renderer.create(
      <RubricItem {...props} />
    ).toJSON()

    let button = explore(tree).selectByID(`rubric-item.edit-comment-${defaultProps.rubricItem.id}`) || {}
    button.props.onPress()

    ActionSheetIOS.showActionSheetWithOptions.mock.calls[0][1](0)
    expect(defaultProps.openCommentKeyboard).toHaveBeenCalledWith(defaultProps.rubricItem.id)
  })

  it('calls show tooltip with the rating description', () => {
    const showToolTip = jest.fn()
    let tree = renderer.create(
      <RubricItem {...defaultProps} showToolTip={showToolTip} />
    ).toJSON()

    let button = explore(tree).selectByID(`rubric-item.points-${defaultProps.rubricItem.ratings[0].id}`) || {}
    button.props.onLongPress(0, { x: 8, y: 9, width: 10, height: 44 })

    expect(showToolTip).toHaveBeenCalledWith(
      { x: 13, y: 9 },
      defaultProps.rubricItem.ratings[0].description
    )
  })
})
