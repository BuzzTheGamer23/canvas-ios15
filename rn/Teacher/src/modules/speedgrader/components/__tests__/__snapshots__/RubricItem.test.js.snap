// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RubricItem changes the currently selected value when a circle is pressed 1`] = `
<View
  style={
    Object {
      "paddingBottom": 8,
      "paddingHorizontal": 16,
      "paddingTop": 16,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": ".SFUIDisplay",
        },
        Object {
          "color": "#2D3B45",
          "fontSize": 16,
        },
        Object {
          "fontWeight": "600",
        },
        Object {
          "fontFamily": ".SFUIDisplay-semibold",
        },
      ]
    }
  >
    A description
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginBottom": 8,
        "marginTop": 12,
      }
    }
  >
    <CircleToggle
      accessibilityLabel="10 — Full Credit"
      on={false}
      onLongPress={[Function]}
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 8,
          "marginRight": 8,
        }
      }
      testID="rubric-item.points-3"
      value={10}
    >
      10
    </CircleToggle>
    <CircleToggle
      accessibilityLabel="5 — Partial Credit"
      on={false}
      onLongPress={[Function]}
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 8,
          "marginRight": 8,
        }
      }
      testID="rubric-item.points-2"
      value={5}
    >
      5
    </CircleToggle>
    <CircleToggle
      accessibilityLabel="0 — No Credit"
      on={false}
      onLongPress={[Function]}
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 8,
          "marginRight": 8,
        }
      }
      testID="rubric-item.points-1"
      value={0}
    >
      0
    </CircleToggle>
    <CircleToggle
      accessibilityLabel="Customize Grade"
      on={false}
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 8,
          "marginRight": 8,
        }
      }
      testID="rubric-item.customize-grade-2"
      value=""
    >
      <Image
        source={1}
      />
    </CircleToggle>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Button
      onPress={[Function]}
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "500",
          "paddingRight": 16,
        }
      }
      testID="rubric-item.add-comment-2"
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#374A59",
                "fontSize": 14,
              },
              Object {
                "fontFamily": ".SFUIDisplay-bold",
              },
              Object {
                "color": "#374A59",
              },
              Object {
                "fontSize": 14,
                "fontWeight": "500",
                "paddingRight": 16,
              },
            ],
            Object {
              "fontFamily": ".SFUIDisplay-medium",
            },
          ]
        }
      >
        Add Comment
      </Text>
    </Button>
    <Button
      onPress={[Function]}
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "500",
          "paddingRight": 16,
        }
      }
      testID="rubric-item.description"
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#374A59",
                "fontSize": 14,
              },
              Object {
                "fontFamily": ".SFUIDisplay-bold",
              },
              Object {
                "color": "#374A59",
              },
              Object {
                "fontSize": 14,
                "fontWeight": "500",
                "paddingRight": 16,
              },
            ],
            Object {
              "fontFamily": ".SFUIDisplay-medium",
            },
          ]
        }
      >
        View long description
      </Text>
    </Button>
  </View>
</View>
`;

exports[`RubricItem renders 1`] = `
<View
  style={
    Object {
      "paddingBottom": 8,
      "paddingHorizontal": 16,
      "paddingTop": 16,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": ".SFUIDisplay",
        },
        Object {
          "color": "#2D3B45",
          "fontSize": 16,
        },
        Object {
          "fontWeight": "600",
        },
        Object {
          "fontFamily": ".SFUIDisplay-semibold",
        },
      ]
    }
  >
    A description
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginBottom": 8,
        "marginTop": 12,
      }
    }
  >
    <CircleToggle
      accessibilityLabel="10 — Full Credit"
      on={false}
      onLongPress={[Function]}
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 8,
          "marginRight": 8,
        }
      }
      testID="rubric-item.points-3"
      value={10}
    >
      10
    </CircleToggle>
    <CircleToggle
      accessibilityLabel="5 — Partial Credit"
      on={false}
      onLongPress={[Function]}
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 8,
          "marginRight": 8,
        }
      }
      testID="rubric-item.points-2"
      value={5}
    >
      5
    </CircleToggle>
    <CircleToggle
      accessibilityLabel="0 — No Credit"
      on={false}
      onLongPress={[Function]}
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 8,
          "marginRight": 8,
        }
      }
      testID="rubric-item.points-1"
      value={0}
    >
      0
    </CircleToggle>
    <CircleToggle
      accessibilityLabel="Customize Grade"
      on={false}
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 8,
          "marginRight": 8,
        }
      }
      testID="rubric-item.customize-grade-2"
      value=""
    >
      <Image
        source={1}
      />
    </CircleToggle>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Button
      onPress={[Function]}
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "500",
          "paddingRight": 16,
        }
      }
      testID="rubric-item.add-comment-2"
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#374A59",
                "fontSize": 14,
              },
              Object {
                "fontFamily": ".SFUIDisplay-bold",
              },
              Object {
                "color": "#374A59",
              },
              Object {
                "fontSize": 14,
                "fontWeight": "500",
                "paddingRight": 16,
              },
            ],
            Object {
              "fontFamily": ".SFUIDisplay-medium",
            },
          ]
        }
      >
        Add Comment
      </Text>
    </Button>
    <Button
      onPress={[Function]}
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "500",
          "paddingRight": 16,
        }
      }
      testID="rubric-item.description"
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#374A59",
                "fontSize": 14,
              },
              Object {
                "fontFamily": ".SFUIDisplay-bold",
              },
              Object {
                "color": "#374A59",
              },
              Object {
                "fontSize": 14,
                "fontWeight": "500",
                "paddingRight": 16,
              },
            ],
            Object {
              "fontFamily": ".SFUIDisplay-medium",
            },
          ]
        }
      >
        View long description
      </Text>
    </Button>
  </View>
</View>
`;

exports[`RubricItem renders an already selected grade 1`] = `
<View
  style={
    Object {
      "paddingBottom": 8,
      "paddingHorizontal": 16,
      "paddingTop": 16,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": ".SFUIDisplay",
        },
        Object {
          "color": "#2D3B45",
          "fontSize": 16,
        },
        Object {
          "fontWeight": "600",
        },
        Object {
          "fontFamily": ".SFUIDisplay-semibold",
        },
      ]
    }
  >
    A description
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginBottom": 8,
        "marginTop": 12,
      }
    }
  >
    <CircleToggle
      accessibilityLabel="10 — Full Credit"
      on={true}
      onLongPress={[Function]}
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 8,
          "marginRight": 8,
        }
      }
      testID="rubric-item.points-3"
      value={10}
    >
      10
    </CircleToggle>
    <CircleToggle
      accessibilityLabel="5 — Partial Credit"
      on={false}
      onLongPress={[Function]}
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 8,
          "marginRight": 8,
        }
      }
      testID="rubric-item.points-2"
      value={5}
    >
      5
    </CircleToggle>
    <CircleToggle
      accessibilityLabel="0 — No Credit"
      on={false}
      onLongPress={[Function]}
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 8,
          "marginRight": 8,
        }
      }
      testID="rubric-item.points-1"
      value={0}
    >
      0
    </CircleToggle>
    <CircleToggle
      accessibilityLabel="Customize Grade"
      on={false}
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 8,
          "marginRight": 8,
        }
      }
      testID="rubric-item.customize-grade-2"
      value=""
    >
      <Image
        source={1}
      />
    </CircleToggle>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Button
      onPress={[Function]}
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "500",
          "paddingRight": 16,
        }
      }
      testID="rubric-item.add-comment-2"
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#374A59",
                "fontSize": 14,
              },
              Object {
                "fontFamily": ".SFUIDisplay-bold",
              },
              Object {
                "color": "#374A59",
              },
              Object {
                "fontSize": 14,
                "fontWeight": "500",
                "paddingRight": 16,
              },
            ],
            Object {
              "fontFamily": ".SFUIDisplay-medium",
            },
          ]
        }
      >
        Add Comment
      </Text>
    </Button>
    <Button
      onPress={[Function]}
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "500",
          "paddingRight": 16,
        }
      }
      testID="rubric-item.description"
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#374A59",
                "fontSize": 14,
              },
              Object {
                "fontFamily": ".SFUIDisplay-bold",
              },
              Object {
                "color": "#374A59",
              },
              Object {
                "fontSize": 14,
                "fontWeight": "500",
                "paddingRight": 16,
              },
            ],
            Object {
              "fontFamily": ".SFUIDisplay-medium",
            },
          ]
        }
      >
        View long description
      </Text>
    </Button>
  </View>
</View>
`;

exports[`RubricItem renders the selected custom grade 1`] = `
<View
  style={
    Object {
      "paddingBottom": 8,
      "paddingHorizontal": 16,
      "paddingTop": 16,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": ".SFUIDisplay",
        },
        Object {
          "color": "#2D3B45",
          "fontSize": 16,
        },
        Object {
          "fontWeight": "600",
        },
        Object {
          "fontFamily": ".SFUIDisplay-semibold",
        },
      ]
    }
  >
    A description
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginBottom": 8,
        "marginTop": 12,
      }
    }
  >
    <CircleToggle
      accessibilityLabel="10 — Full Credit"
      on={false}
      onLongPress={[Function]}
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 8,
          "marginRight": 8,
        }
      }
      testID="rubric-item.points-3"
      value={10}
    >
      10
    </CircleToggle>
    <CircleToggle
      accessibilityLabel="5 — Partial Credit"
      on={false}
      onLongPress={[Function]}
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 8,
          "marginRight": 8,
        }
      }
      testID="rubric-item.points-2"
      value={5}
    >
      5
    </CircleToggle>
    <CircleToggle
      accessibilityLabel="0 — No Credit"
      on={false}
      onLongPress={[Function]}
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 8,
          "marginRight": 8,
        }
      }
      testID="rubric-item.points-1"
      value={0}
    >
      0
    </CircleToggle>
    <CircleToggle
      accessibilityLabel="Customize Grade"
      on={true}
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 8,
          "marginRight": 8,
        }
      }
      testID="rubric-item.customize-grade-2"
      value="12"
    >
      12
    </CircleToggle>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Button
      onPress={[Function]}
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "500",
          "paddingRight": 16,
        }
      }
      testID="rubric-item.add-comment-2"
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#374A59",
                "fontSize": 14,
              },
              Object {
                "fontFamily": ".SFUIDisplay-bold",
              },
              Object {
                "color": "#374A59",
              },
              Object {
                "fontSize": 14,
                "fontWeight": "500",
                "paddingRight": 16,
              },
            ],
            Object {
              "fontFamily": ".SFUIDisplay-medium",
            },
          ]
        }
      >
        Add Comment
      </Text>
    </Button>
    <Button
      onPress={[Function]}
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "500",
          "paddingRight": 16,
        }
      }
      testID="rubric-item.description"
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#374A59",
                "fontSize": 14,
              },
              Object {
                "fontFamily": ".SFUIDisplay-bold",
              },
              Object {
                "color": "#374A59",
              },
              Object {
                "fontSize": 14,
                "fontWeight": "500",
                "paddingRight": 16,
              },
            ],
            Object {
              "fontFamily": ".SFUIDisplay-medium",
            },
          ]
        }
      >
        View long description
      </Text>
    </Button>
  </View>
</View>
`;

exports[`RubricItem will not show the add comment button and will show the comment if there is one 1`] = `
<View
  style={
    Object {
      "paddingBottom": 8,
      "paddingHorizontal": 16,
      "paddingTop": 16,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": ".SFUIDisplay",
        },
        Object {
          "color": "#2D3B45",
          "fontSize": 16,
        },
        Object {
          "fontWeight": "600",
        },
        Object {
          "fontFamily": ".SFUIDisplay-semibold",
        },
      ]
    }
  >
    A description
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginBottom": 8,
        "marginTop": 12,
      }
    }
  >
    <CircleToggle
      accessibilityLabel="10 — Full Credit"
      on={false}
      onLongPress={[Function]}
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 8,
          "marginRight": 8,
        }
      }
      testID="rubric-item.points-3"
      value={10}
    >
      10
    </CircleToggle>
    <CircleToggle
      accessibilityLabel="5 — Partial Credit"
      on={false}
      onLongPress={[Function]}
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 8,
          "marginRight": 8,
        }
      }
      testID="rubric-item.points-2"
      value={5}
    >
      5
    </CircleToggle>
    <CircleToggle
      accessibilityLabel="0 — No Credit"
      on={false}
      onLongPress={[Function]}
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 8,
          "marginRight": 8,
        }
      }
      testID="rubric-item.points-1"
      value={0}
    >
      0
    </CircleToggle>
    <CircleToggle
      accessibilityLabel="Customize Grade"
      on={false}
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 8,
          "marginRight": 8,
        }
      }
      testID="rubric-item.customize-grade-2"
      value=""
    >
      <Image
        source={1}
      />
    </CircleToggle>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Button
      onPress={[Function]}
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "500",
          "paddingRight": 16,
        }
      }
      testID="rubric-item.description"
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#374A59",
                "fontSize": 14,
              },
              Object {
                "fontFamily": ".SFUIDisplay-bold",
              },
              Object {
                "color": "#374A59",
              },
              Object {
                "fontSize": 14,
                "fontWeight": "500",
                "paddingRight": 16,
              },
            ],
            Object {
              "fontFamily": ".SFUIDisplay-medium",
            },
          ]
        }
      >
        View long description
      </Text>
    </Button>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingTop": 16,
      }
    }
  >
    <View
      style={
        Object {
          "maxWidth": 300,
        }
      }
    >
      <Image
        capInsets={
          Object {
            "bottom": 16,
            "left": 18,
            "right": 18,
            "top": 24,
          }
        }
        resizeMode="stretch"
        source={1}
        style={
          Object {
            "bottom": 0,
            "height": null,
            "left": 0,
            "margin": 0,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": -4,
            "width": null,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
                "paddingBottom": 8,
                "paddingHorizontal": 12,
                "paddingTop": 12,
              },
              Object {
                "color": "#2D3B45",
              },
            ],
            Object {
              "fontFamily": ".SFUIDisplay",
            },
          ]
        }
      >
        A comment
      </Text>
    </View>
    <Button
      onPress={[Function]}
      style={
        Object {
          "color": "#73818C",
          "fontSize": 16,
          "fontWeight": "600",
        }
      }
      testID="rubric-item.edit-comment-2"
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#374A59",
                "fontSize": 14,
              },
              Object {
                "fontFamily": ".SFUIDisplay-bold",
              },
              Object {
                "color": "#374A59",
              },
              Object {
                "color": "#73818C",
                "fontSize": 16,
                "fontWeight": "600",
              },
            ],
            Object {
              "fontFamily": ".SFUIDisplay-semibold",
            },
          ]
        }
      >
        Edit
      </Text>
    </Button>
  </View>
</View>
`;
